metadata-service/factories/src/test/java/com/linkedin/metadata/boot/steps/RestoreGlossaryIndicesTest.java

```diff
@@ -0,0 +1,158 @@
+            Collections.singleton(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME)))
+        .thenReturn(termInfoResponses);
+
+    //  Mock nodeInfoResponses and getting aspectSpec
+    Map<String, EnvelopedAspect> nodeInfoAspects = new HashMap<>();
+    nodeInfoAspects.put(Constants.GLOSSARY_NODE_INFO_ASPECT_NAME, new EnvelopedAspect().setValue(new Aspect(new GlossaryNodeInfo().setName("test").data())));
+    Map<Urn, EntityResponse> nodeInfoResponses = new HashMap<>();
+    nodeInfoResponses.put(glossaryNodeUrn, new EntityResponse().setUrn(glossaryNodeUrn).setAspects(new EnvelopedAspectMap(nodeInfoAspects)));
+    Mockito.when(mockSearchService.search(Constants.GLOSSARY_NODE_ENTITY_NAME, "", null, null, 0, 1000))
+        .thenReturn(new SearchResult().setNumEntities(1).setEntities(new SearchEntityArray(ImmutableList.of(new SearchEntity().setEntity(glossaryNodeUrn)))));
+    Mockito.when(mockService.getEntitiesV2(
+        Constants.GLOSSARY_NODE_ENTITY_NAME,
+            new HashSet<>(Collections.singleton(glossaryNodeUrn)),
+            Collections.singleton(Constants.GLOSSARY_NODE_INFO_ASPECT_NAME)
+        ))
+        .thenReturn(nodeInfoResponses);
+
+    EntitySpec entitySpec = Mockito.mock(EntitySpec.class);
+    AspectSpec aspectSpec = Mockito.mock(AspectSpec.class);
+    //  Mock for Terms
+    Mockito.when(mockRegistry.getEntitySpec(Constants.GLOSSARY_TERM_ENTITY_NAME)).thenReturn(entitySpec);
+    Mockito.when(entitySpec.getAspectSpec(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME)).thenReturn(aspectSpec);
+    //  Mock for Nodes
+    Mockito.when(mockRegistry.getEntitySpec(Constants.GLOSSARY_NODE_ENTITY_NAME)).thenReturn(entitySpec);
+    Mockito.when(entitySpec.getAspectSpec(Constants.GLOSSARY_NODE_INFO_ASPECT_NAME)).thenReturn(aspectSpec);
+
+    RestoreGlossaryIndices restoreIndicesStep = new RestoreGlossaryIndices(mockService, mockSearchService, mockRegistry);
+    restoreIndicesStep.execute();
+
+
+    Mockito.verify(mockService, Mockito.times(2)).ingestProposal(
+        Mockito.any(MetadataChangeProposal.class),
+        Mockito.any(AuditStamp.class)
+        );
+    Mockito.verify(mockService, Mockito.times(1)).produceMetadataChangeLog(
+        Mockito.eq(glossaryTermUrn),
+        Mockito.eq(Constants.GLOSSARY_TERM_ENTITY_NAME),
+        Mockito.eq(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME),
+        Mockito.eq(aspectSpec),
+        Mockito.eq(null),
+        Mockito.any(),
+        Mockito.eq(null),
+        Mockito.eq(null),
+        Mockito.any(),
+        Mockito.eq(ChangeType.RESTATE)
+    );
+    Mockito.verify(mockService, Mockito.times(1)).produceMetadataChangeLog(
+        Mockito.eq(glossaryNodeUrn),
+        Mockito.eq(Constants.GLOSSARY_NODE_ENTITY_NAME),
+        Mockito.eq(Constants.GLOSSARY_NODE_INFO_ASPECT_NAME),
+        Mockito.eq(aspectSpec),
+        Mockito.eq(null),
+        Mockito.any(),
+        Mockito.eq(null),
+        Mockito.eq(null),
+        Mockito.any(),
+        Mockito.eq(ChangeType.RESTATE)
```

Comment:
Was this working? Are we producing a RESTATE change?

Reply:
we are for the `produceMetadataChangelog` calls (which is what this is testing here)